### PR TITLE
Add a note to changelog about redis usage

### PR DESCRIPTION
**Pull request by erikjohnston**
c.f. #9198

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Unreleased
+==========
+
+Note that this release includes a change in Synapse to use Redis as a cache, as well as a pub/sub mechanism, if Redis support is enabled. No action is needed by server administrators.
```
Comment (erikjohnston) [inline CHANGES.md:4]: I don't know if its worth saying that we don't expect much increase in memory usage? Matrix.org is currently running at ~20MB.

**Review by clokep — APPROVED** [review]
This seems OK. I do wonder if people will have questions about what is being cached? 🤷

File: CHANGES.md (hunk 1 above)
Comment (clokep) [reply CHANGES.md:4]: It might be worth mentioning that we don't expect the Redis usage to dramatically increase.

**Comment by clokep** [comment]
Should we update anything in the workers doc: https://github.com/matrix-org/synapse/blob/master/docs/workers.md#main-processworker-communication (or the arch diagram in it).

File: CHANGES.md (hunk 1 above)
Comment (erikjohnston) [reply CHANGES.md]: Done

**Comment by erikjohnston** [comment]
> This seems OK. I do wonder if people will have questions about what is being cached?

I think at that point they can look at the PR, I don't think its useful to spell it out anywhere, especially since I imagine we'll be caching more stuff in Redis as time goes by.

**Pull request merged**